src/visualisation/CSVWriter.cpp

```diff
@@ -142,65 +82,52 @@ void CSVWriter::csv_write(const char *filename, real **vars, int size_vars, cons
     real dy = domain->get_dy();
     real dz = domain->get_dz();
 
-    const char *coord_names[] = {"i", "j", "k", "index", "x-coords (m)", "y-coords (m)", "z-coords (m)"};
+    std::vector<std::string> coord_names = {"i", "j", "k", "index",
+                                            "x-coords (m)", "y-coords (m)", "z-coords (m)"};
 
     auto *x_centres = new float[size];
     auto *y_centres = new float[size];
     auto *z_centres = new float[size];
```

Comment:
we should change the centre values to real as well, for consistency reason. Guess some old code which wasn't changed yet

Reply:
updatet in 70b52fc